homeassistant/components/hassio/handler.py

```diff
@@ -46,10 +46,14 @@ def api_data(funct):
 
     async def _wrapper(*argv, **kwargs):
         """Wrap function."""
-        data = await funct(*argv, **kwargs)
-        if data["result"] == "ok":
-            return data["data"]
-        raise HassioAPIError(data["message"])
+        while True:
+            data = await funct(*argv, **kwargs)
+            if data["result"] == "ok":
+                return data["data"]
+            if data["message"] == "System is not ready with state: setup":
```

Comment:
Ideally we check for an http status here and match it as well

Reply:
Agreed, supervisor should actually respond with a suitable error code e.g. 503 not with a 400 client error. We should not need to rely on distinguishing this by a message string which might change arbitrarily.